deepchecks/recommender/checks/date_train_test_leakage_overlap.py

```diff
@@ -0,0 +1,171 @@
+from deepchecks.tabular.utils.task_type import TaskType
+import plotly.graph_objects as go
+__all__ = ['DateTrainTestLeakageOverlap']
+
+
+class DateTrainTestLeakageOverlap(TrainTestCheck):
+    """Check test data that is dated earlier than the latest date in train. \
+       The validation per user (validation_per_user=True) use the "N leave out" validation schema which is largely used in recommender systems \
+        to evaluate the performance of the model by leaving out a certain number of items or interactions for each user during training. \
+        In the context of validating by using the last history of each user, the leave-n-out approach involves withholding the most recent n items \
+        or interactions of each user from the training data and using them as a validation set. \
+        The model is trained on the remaining historical data for each user and then evaluated on the held-out interactions. \
+        Whereas, when time validation is used (validation_per_user=True), we will validate in the last time window of our data.
+
+    Parameters
+    ----------
+    n_samples : int , default: 1_000_000
+        number of samples to use for this check.
+    random_state : int, default: 42
+        random seed for all check internals.
+    validation_per_user : bool, default: True
+         if set to True :evaluate the performance of the model by leaving out N items for each user.
+    """
+
+    def __init__(
+        self,
+        n_samples: int = 1_000_000,
+        random_state: int = 42,
+        validation_per_user: bool = False,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.n_samples = n_samples
+        self.random_state = random_state
+        self.validation_per_user = validation_per_user
+
+    def run_logic(self, context: Context) -> CheckResult:
+        """Run check.
+
+        Returns
+        -------
+        CheckResult
+            value is the ratio of date leakage.
+            data is html display of the checks' textual result.
+
+        Raises
+        ------
+        DeepchecksValueError
+            If one of the datasets is not a Dataset instance with an date
+        """
+        train_dataset = context.train.sample(self.n_samples, random_state=self.random_state)
+        test_dataset = context.test.sample(self.n_samples, random_state=self.random_state)
+        display = []
+
+        train_dataset.assert_datetime()
+        train_date = train_dataset.datetime_col
+        test_date = test_dataset.datetime_col
+        if self.validation_per_user==False:
+            max_train_date = max(train_date)
+            min_test_date = min(test_date)
+            dates_leaked = sum(date < max_train_date for date in test_date)
+            if dates_leaked > 0:
+                leakage_ratio = dates_leaked / test_dataset.n_samples
+                return_value = leakage_ratio
+                text = f'{format_percent(leakage_ratio)} of test data samples are in the date range ' \
```

Comment:
This information (min_test_date and max_train_date) should probably be in the result value and not in the display, since you've now created a display that is better than just printing text.

Reply:
Fixed.

Reply:
fixed https://github.com/deepchecks/deepchecks/pull/2610/commits/d97c7e0ad0c40b007c04a9f2c9920691c42d48df

<img width="850" alt="Capture d’écran 2023-07-07 à 4 07 57 PM" src="https://github.com/deepchecks/deepchecks/assets/55285736/76f97943-0f57-4c24-a53c-5e289765a411">